mondey_backend/tests/routers/test_auth.py

```diff
@@ -0,0 +1,41 @@
+import smtplib
+from email.message import EmailMessage
+
+import pytest
+from fastapi.testclient import TestClient
+
+
+class SMTPMock:
+    last_message: EmailMessage | None = None
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def send_message(self, msg: EmailMessage, **kwargs):
+        SMTPMock.last_message = msg
+
+
+@pytest.fixture
+def smtp_mock(monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setattr(smtplib, "SMTP", SMTPMock)
+    return SMTPMock
+
+
+def test_register_new_user(public_client: TestClient, smtp_mock: SMTPMock):
```

Comment:
do we need a test for failed verification?

Reply:
yes true, will add one